### PR TITLE
Rewards oracle wrapper contract

### PR DESCRIPTION
**Pull request by ThornM9**

### CODE REVIEW

File: programs/rewards-oracle/src/instructions/set_current_rewards_wrapper_v0.rs [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+  pub entity_key: Vec<u8>,
+  pub oracle_index: u16,
+  pub current_rewards: u64,
+}
+
+#[derive(Accounts)]
+#[instruction(args: SetCurrentRewardsWrapperArgsV0)]
+pub struct SetCurrentRewardsWrapperV0<'info> {
+  #[account(mut)]
+  pub payer: Signer<'info>,
```
Comment (ChewingGlass) [inline programs/rewards-oracle/src/instructions/set_current_rewards_wrapper_v0.rs:20]: Isn't actually payer, I'd call this oracle.

File: programs/lazy-distributor/src/lib.rs [hunk 2]
```diff
@@ -11,6 +11,15 @@ pub mod token_metadata;
 pub use instructions::*;
 pub use state::*;
 
+#[derive(Clone)]
+pub struct LazyDistributor;
+
+impl anchor_lang::Id for LazyDistributor {
```
Comment (ChewingGlass) [inline programs/lazy-distributor/src/lib.rs:17]: Shouldn't be needed, anchor exports this doesn't it?

File: programs/rewards-oracle/src/state.rs [hunk 3]
```diff
@@ -0,0 +1,18 @@
+use anchor_lang::prelude::*;
+
+#[account]
+#[derive(Default)]
+pub struct PriceOracleV0 {
```
Comment (ChewingGlass) [inline programs/rewards-oracle/src/state.rs:5]: This shouldn't be needed or used right?

File: programs/lazy-distributor/src/lib.rs (hunk 2 above)
Comment (ThornM9) [reply programs/lazy-distributor/src/lib.rs:17]: Anchor doesn't export it, this code block is in all our programs that are CPI'd into

File: programs/rewards-oracle/src/state.rs (hunk 3 above)
Comment (ThornM9) [reply programs/rewards-oracle/src/state.rs:5]: Yep, updated

File: programs/rewards-oracle/src/instructions/set_current_rewards_wrapper_v0.rs (hunk 1 above)
Comment (ThornM9) [reply programs/rewards-oracle/src/instructions/set_current_rewards_wrapper_v0.rs:20]: True

**Review by ChewingGlass — APPROVED** [review]

**Pull request merged**